File: javascript/leetcode/1051. Height Checker.js
/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function (heights) {
  const dupe = [...heights];

  dupe.sort((a, b) => a - b);

  let ans = 0;

  for (let i = 0; i < heights.length; ++i) {
    if (dupe[i] !== heights[i]) ++ans;
  }

  return ans;
};

File: c++/leetcode/1051. Height Checker.cpp
class Solution
{
public:
  int heightChecker(vector<int> &heights)
  {
    auto dupe = heights;
    sort(dupe.begin(), dupe.end());

    int ans = 0;

    for (int i = 0; i < heights.size(); ++i)
    {
      if (dupe[i] != heights[i])
        ++ans;
    }

    return ans;
  }
};